tests/internal/testopenai/requests.go

```diff
@@ -385,4 +413,163 @@ var requestBodies = map[Cassette]*openai.ChatCompletionRequest{
+						},
+					},
+				},
+			},
+		},
+	},
+	CassetteChatTextToAudio: {
+		Model: openai.ModelGPT4oMiniAudioPreview,
+		Messages: []openai.ChatCompletionMessageParamUnion{
+			{
+				Type: openai.ChatMessageRoleUser,
+				Value: openai.ChatCompletionUserMessageParam{
+					Role: openai.ChatMessageRoleUser,
+					Content: openai.StringOrUserRoleContentUnion{
+						Value: "Say a single short 'beep' sound, as brief as possible.",
+					},
+				},
+			},
+		},
+		Audio: &openai.ChatCompletionAudioParam{
+			Format: openai.ChatCompletionAudioFormatOpus,
+			Voice:  openai.ChatCompletionAudioVoiceAlloy,
+		},
+		Modalities: []openai.ChatCompletionModality{
+			openai.ChatCompletionModalityText,
+			openai.ChatCompletionModalityAudio,
+		},
+	},
+	CassetteChatDetailedUsage: {
+		Model: openai.ModelGPT41Nano,
+		Messages: []openai.ChatCompletionMessageParamUnion{
+			{
+				Type: openai.ChatMessageRoleDeveloper,
+				Value: openai.ChatCompletionDeveloperMessageParam{
+					Role: openai.ChatMessageRoleDeveloper,
+					Content: openai.StringOrArray{
+						Value: "You are a poetry assistant. Write a haiku when asked.",
+					},
+				},
+			},
+			{
+				Type: openai.ChatMessageRoleUser,
+				Value: openai.ChatCompletionUserMessageParam{
+					Role: openai.ChatMessageRoleUser,
+					Content: openai.StringOrUserRoleContentUnion{
+						Value: "Write a haiku about OpenTelemetry tracing.",
+					},
+				},
+			},
+		},
+		Temperature: ptr.To(0.8),
+		MaxTokens:   ptr.To[int64](100),
+	},
+	CassetteChatStreamingDetailedUsage: {
+		Model: openai.ModelGPT41Nano,
+		Messages: []openai.ChatCompletionMessageParamUnion{
+			{
+				Type: openai.ChatMessageRoleUser,
+				Value: openai.ChatCompletionUserMessageParam{
+					Role: openai.ChatMessageRoleUser,
+					Content: openai.StringOrUserRoleContentUnion{
+						Value: "Say hello",
+					},
+				},
+			},
+		},
+		Stream: true,
+		StreamOptions: &openai.StreamOptions{
+			IncludeUsage: true,
+		},
+		MaxTokens: ptr.To[int64](10),
+	},
+	CassetteChatTextToImageTool: {
+		Model: openai.ModelGPT41Nano,
+		Messages: []openai.ChatCompletionMessageParamUnion{
+			{
+				Type: openai.ChatMessageRoleSystem,
+				Value: openai.ChatCompletionSystemMessageParam{
+					Role: openai.ChatMessageRoleSystem,
+					Content: openai.StringOrArray{
+						Value: "You are an AI assistant that generates simple, sketch-style images with minimal detail. When asked to generate an image, create it with low quality settings for cost efficiency.",
+					},
+				},
+			},
+			{
+				Type: openai.ChatMessageRoleUser,
+				Value: openai.ChatCompletionUserMessageParam{
+					Role: openai.ChatMessageRoleUser,
+					Content: openai.StringOrUserRoleContentUnion{
+						Value: "Draw a simple, minimalist image of a cute cat playing with a ball of yarn in a sketch style.",
+					},
+				},
+			},
+		},
+		Tools: []openai.Tool{
+			{
+				Type: openai.ToolTypeFunction,
+				Function: &openai.FunctionDefinition{
+					Name:        "generate_image",
```

Comment:
trivia: strangest lack of parity is that you can do text to audio and audio to text, and you can do image to text, but if you want to do text to image, it requires use of a special tool call!